src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -856,8 +859,7 @@ public ParseResult ParseAndBind(ScriptFile scriptFile, ConnectionInfo connInfo)
                     try
                     {
                         scriptInfo.ConnectionKey = this.BindingQueue.AddConnectionContext(info, "languageService");
-                        scriptInfo.IsConnected = true;
-
+                        scriptInfo.IsConnected = this.BindingQueue.IsBindingContextConnected(scriptInfo.ConnectionKey);
```

Comment:
we should only be queuing tasks that are connected, since the whole point is to share connections.  Is this flag to track the current connection state for when connections drops offline?

Reply:
This is to set the correct value - the `BindingQueue.AddConnectionContext` does not throw on failure to connect and internally marks the context as not connected. However the scriptInfo assumed that the connection occurred. With this change, the intellisense code should be able to tell that the script isn't connected and return the default completions without ever interacting with the binding queue.